Explorer/Assets/DCL/PerformanceAndDiagnostics/Profiling/Profiler.cs

```diff
@@ -74,8 +77,29 @@ public void Dispose()
             return new FrameTimeStats(minFrameTime, maxFrameTime, hiccupCount);
         }
 
-        public AnalyticsFrameTimeReport? GetMainThreadFramesNs(int[] percentile) =>
-            GetFrameStatsWithPercentiles(mainThreadTimeRecorder, percentile);
+        private static float GetRecorderSamplesSum(ProfilerRecorder recorder)
+        {
+            int samplesCount = recorder.Capacity;
+
+            if (samplesCount == 0)
+                return 0;
+
+            float r = 0;
+
+            unsafe
+            {
+                ProfilerRecorderSample* samples = stackalloc ProfilerRecorderSample[samplesCount];
```

Comment:
It's totally fine but you could use `Span` to avoid `unsafe` code. I doubt there unsafe code here brought any significant benefits

Reply:
Unity in their documentation and tutorials always used `unsafe` for some reason [link1](https://docs.unity3d.com/ScriptReference/Unity.Profiling.ProfilerRecorder.CopyTo.html), [link2](https://docs.unity3d.com/2023.2/Documentation/ScriptReference/Unity.Profiling.ProfilerRecorder.html). But `Span` is quite close in performance, agree

Reply:
turned out that Unity implementation of `profilerRecorder.CopyTo`  has only 1 safe version and it is with the `List`. I refactored to use it. 

Unity API for copying `profilerRecorder` :

```
 public void CopyTo(List<ProfilerRecorderSample> outSamples, bool reset = false)
    {
      if (outSamples == null)
        throw new ArgumentNullException(nameof (outSamples));
      this.CheckInitializedAndThrow();
      ProfilerRecorder.CopyTo_List(this, outSamples, reset);
    }

    public unsafe int CopyTo(ProfilerRecorderSample* dest, int destSize, bool reset = false)
    {
      this.CheckInitializedWithParamsAndThrow(dest);
      return ProfilerRecorder.CopyTo_Pointer(this, dest, destSize, reset);
    }

    public unsafe ProfilerRecorderSample[] ToArray()
    {
      this.CheckInitializedAndThrow();
      int count = this.Count;
      ProfilerRecorderSample[] array = new ProfilerRecorderSample[count];
      fixed (ProfilerRecorderSample* outSamples = array)
        ProfilerRecorder.CopyTo_Pointer(this, outSamples, count, false);
      return array;
    }
```